lib/resty/etcd/v3.lua

```diff
@@ -125,6 +126,15 @@ local function _request_uri(self, method, uri, opts, timeout, ignore_auth)
         end
     end
 
+    if self.extra_headers and type(self.extra_headers) == "table" then
+        for key, value in pairs(self.extra_headers) do
+            if key ~= "Authorization" then
```

Comment:
Setting for Content-Length, Transfer-Encoding, Connection, Upgrade should be ignored.

Reply:
update